cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java

```diff
@@ -176,6 +168,9 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
             List<String> cmdsList = new ArrayList<>();
             if (xml != null && xml.trim().length() > 0) {
+                if (!xml.endsWith(".xml")) {
+                    throw new CommandFormatException("The --server-config filename must end with .xml.");
+                }
```

Comment:
Yep, I'd already opened https://issues.jboss.org/browse/WFCORE-991 for this. There's some suspicious code in ConfiguratioinFile.addSuffixToFile(historyBase, INITIAL) assuming that the rightmost . is part of a file suffix, and without it being present it appends the config name to the next . it encounters to create the path for the initial config file backup. This is almost certainly a directory that doesn't exist.